src/actionTypes.js

```diff
@@ -211,7 +215,7 @@ export type MessageFetchCompleteAction = {
   numAfter: number,
 };
 
-export type FetchMessagesAction = any;
+export type FetchMessagesAction = (dispatch: Dispatch, getState: GetState) => Promise<void> | void;
```

Comment:
The definition you are providing is actually an action creator.
Have a look at the distinction between actions and action creators (the second creates the action 😄 )
https://redux.js.org/basics/actions

Reply:
What seems to be confusing in this case is that the name is actually incorrect, and likely I did introduce that error :)
In `fetchActions` the return type of `fetchMessages` is not `FetchMessagesAction`
Compare that to `messageFetchComplete` where the returned type is indeed an action.

Maybe the right choice, for now, is to remove the `FetchMessagesAction` type.

Reply:
Shall I add this type as `ActionCreator` and replace the instances of `FetchMessagesAction` with `ActionCreator`?
Or the preferred way would be to remove the type entirely and define it inline?

Reply:
Have a look at the `redux_v4.x.x.js` file which contains the definitions for Redux (coming from the FlowTyped project).

It already contains an `ActionCreator` type which you will want to reuse.

We haven't used it anywhere so you can't look in our source code, but Googling for examples will probably work fine.

Reply:
On it. Thanks a lot. Will implement as soon as i can.

Reply:
@borisyankov  Sorry for getting back so late but I needed some clarification here:
In `redux_v4.x.x.js`, `ActionCreator` is defined as follows: `ActionCreator<A, B> = (...args: Array<B>) => A;`. I had a hard time interpreting `...args: Array<B>` because we aren't passing any array here. 
I tried to just fit my above definition of ActionCreator into this and this is what I thought worked: `FetchMessagesAction = ActionCreator<void | Promise<void>, GetState | Dispatch>;` and flow gave no errors but then i went ahead and made a small change `FetchMessagesAction = ActionCreator<void | Promise<void>, GetState>;` removing `Dispatch` and flow still didn't give me an error which led me to conclude that this implementation of mine might be wrong. Seeking some guidance on this.

Reply:
Fixed in de70ffd thanks to @gnprice